src/algorithms/dbscan.py

```diff
@@ -0,0 +1,53 @@
+import numpy as np
+from sklearn.cluster import DBSCAN as _DBSCAN  # type: ignore
+
+from src.utils import reshape_to_2d
+
+from .base_clustering import BaseClustering
+
+
+class DBSCAN(BaseClustering):
+    """DBSCAN clustering algorithm implementation.
+
+    Args:
+        kwargs (dict): Keyword arguments to initialize the DBSCAN algorithm.
+
+    Attributes:
+        algorithm (_DBSCAN): Instance of the DBSCAN algorithm.
+
+    """
+
+    def __init__(self, kwargs: dict):
+        """Initialize the DBSCAN clustering algorithm.
+
+        Args:
+            kwargs (dict): Keyword arguments to initialize the DBSCAN algorithm.
+
+        """
+        self.algorithm: _DBSCAN = _DBSCAN(**kwargs)
```

Comment:
you do not need to save the algorithm method, it should be hardcoded in this class implementation

Reply:
Do you mean refactor the process method to instantiate the DBSCAN algorithm directly within the method instead of storing it as an attribute?

```

    def process(self, data: np.ndarray) -> np.ndarray:
        """Process the input data using the DBSCAN algorithm.

        Args:
            data (np.ndarray): Input data to be clustered.

        Returns:
            np.ndarray: Array of cluster labels assigned to each data point.
        """
        algorithm = _DBSCAN(eps=self.eps, min_samples=self.min_samples)
        return algorithm.fit_predict(data)
```


Instantiating the DBSCAN algorithm every time `process` is called, may be less efficient if `process` is called frequently with the same parameters.

Reply:
Here is another solution:

```
class DBSCAN(BaseClustering):
    """DBSCAN clustering algorithm implementation."""

    def __init__(self, eps: float, min_samples: int):
        """
        Initialize the DBSCAN clustering algorithm.

        Args:
            eps (float): The maximum distance between two samples for one to be considered
                as in the neighborhood of the other.
            min_samples (int): The number of samples (or total weight) in a neighborhood for
                a point to be considered as a core point.
        """
        self.algorithm = _DBSCAN(eps=eps, min_samples=min_samples)

    def process(self, data: np.ndarray) -> np.ndarray:
        """Process the input data using DBSCAN clustering algorithm.

        Args:
            data (np.ndarray): Input data to be clustered.

        Returns:
            np.ndarray: Array of cluster labels assigned to each data point.
        """
        return self.algorithm.fit_predict(data)
```